grpc/controlproxy/controlproxy.go

```diff
@@ -0,0 +1,240 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+
+Copyright 2015-2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package controlproxy
+
+import (
+	"errors"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/intelsdi-x/snap/core"
+	"github.com/intelsdi-x/snap/core/ctypes"
+	"github.com/intelsdi-x/snap/core/serror"
+	"github.com/intelsdi-x/snap/grpc/common"
+	"github.com/intelsdi-x/snap/grpc/controlproxy/rpc"
+	"github.com/intelsdi-x/snap/pkg/rpcutil"
+)
+
+// Implements managesMetrics interface provided by scheduler and
+// proxies those calls to the grpc client.
+type ControlProxy struct {
+	Client rpc.MetricManagerClient
+}
+
+func New(addr string, port int) (ControlProxy, error) {
+	c, err := NewClient(addr, port)
+	if err != nil {
+		return ControlProxy{}, err
+	}
+	return ControlProxy{Client: c}, nil
+}
+
+// NewClient returns a control RPC Client.
+func NewClient(addr string, port int) (rpc.MetricManagerClient, error) {
```

Comment:
I think right now it is exported only to be used in control/control_grpc_server_test.go.